LX/src/main/java/heronarts/lx/LXSnapshotController.java

```diff
@@ -0,0 +1,161 @@
+package heronarts.lx;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeMap;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import heronarts.lx.osc.OscMessage;
+
+public class LXSnapshotController extends LXComponent {
+
+  private Integer nextShapshotID = 1;
+
+  private final List<LXSnapshot> snapshots = new ArrayList<LXSnapshot>();
+  private final TreeMap<Integer, LXSnapshot> snapshotsDict = new TreeMap<Integer, LXSnapshot>();
+
+  public interface SnapshotListener {
+    public void snapshotAdded(LXSnapshotController controller, LXSnapshot snapshot);
+    public void snapshotRemoved(LXSnapshotController controller, LXSnapshot snapshot);
+  }
+
+  private final List<SnapshotListener> snapshotListeners = new ArrayList<SnapshotListener>();
+
+  public LXSnapshotController(LX lx) {
+    super(lx);
+    this.label.setDescription("Snapshots");
```

Comment:
I think you might mean setValue() here? But better to use a different super() constructor:

super(lx, "Snapshots");

That sets default label.